src/OnlineRetailPortal.Core/Models/Category.cs

```diff
@@ -1,19 +1,31 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using OnlineRetailPortal.Contracts;
+using OnlineRetailPortal.Core.Object_Factory;
+using OnlineRetailPortal.Mock;
 
-namespace OnlineRetailPortal.Core.Models
+namespace OnlineRetailPortal.Core
 {
-    public enum Category
+    public class Category
     {
-        Properties,
-        Cars,
-        Furniture,
-        Mobiles,
-        Bikes,
-        Books,
-        Fashions,
-        Electronics,
-        Others
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        private ICategoryStore _categoryStore;
+        
+        public Category()
+        {
+            CategoryObjectFactory categoryObjectFactory = new CategoryObjectFactory();
+            _categoryStore = categoryObjectFactory.GetCategoryStore();
```

Comment:
this should not be a part of this constructor. Every time category instance is created, instance for these are created as well, which is not required. Try to implement this on the service layer

Reply:
I inject Dependency instead of using Object Factory, check if it's okay